subsys/bluetooth/host/gatt.c

```diff
@@ -687,11 +717,19 @@ ssize_t bt_gatt_attr_write_ccc(struct bt_conn *conn,
 		gatt_ccc_changed(attr, ccc);
 
 #if defined(CONFIG_BT_SETTINGS_CCC_STORE_ON_WRITE)
-		if (atomic_cas(&gatt_ccc_store.scheduled, false, true) &&
-		    bt_addr_le_is_bonded(conn->id, &conn->le.dst)) {
-			gatt_ccc_store.conn = bt_conn_ref(conn);
+		if (gatt_ccc_conn_is_queued(conn)) {
+			BT_DBG("Already queued");
+			return BT_GATT_ERR(BT_ATT_ERR_PROCEDURE_IN_PROGRESS);
```

Comment:
Not sure if the debug log above is really necessary, also I don't think we should return an error here since it just means the cache is not updated but it already written and the notification/indication should be on already.

Reply:
You are right, here we should just omit writing the value again, not return error. Will fix

Reply:
Fixed